cmd/agent/main.go

```diff
@@ -222,54 +222,73 @@ func main() {
+			log.Fatalw("API key is missing from loaded config")
+		}
 
-	platform, err := transport.NewPlatformGRPC(transportConf)
-	if err != nil {
-		log.Fatalw("transport initialize error", zap.Error(err))
-	}
+		transportConf := transport.PlatformGRPCConf{
+			UUID:            global.AgentHostname,
+			APIKey:          global.AgentConf.Platform.APIKey,
+			TransmitTimeout: global.AgentConf.Platform.TransportTimeout,
+			URL:             global.AgentConf.Platform.Addr,
+		}
 
-	// initialize the buffer for temporary in-memory caching of collected data
-	// and its controller for maintaining and accessing the buffer.
-	bufCtrlConf := buf.ControllerConf{
-		BufLenLimit:         global.AgentConf.Platform.BatchN,
-		BufDrainFreq:        global.AgentConf.Platform.MaxPublishInterval,
-		OnBufRemoveCallback: platform.PublishFunc,
-		MaxHeapAllocBytes:   global.AgentConf.Buffer.MaxHeapAlloc,
-		MinBufSize:          global.AgentConf.Buffer.MinBufferSize,
-	}
+		platform, err := transport.NewPlatformGRPC(transportConf)
+		if err != nil {
+			log.Fatalw("transport initialize error", zap.Error(err))
+		}
 
-	buffer := buf.NewPriorityBuffer(global.AgentConf.Buffer.TTL)
-	bufCtrl := buf.NewController(bufCtrlConf, buffer)
+		// initialize the buffer for temporary in-memory caching of collected data
+		// and its controller for maintaining and accessing the buffer.
+		bufCtrlConf := buf.ControllerConf{
+			BufLenLimit:         global.AgentConf.Platform.BatchN,
+			BufDrainFreq:        global.AgentConf.Platform.MaxPublishInterval,
+			OnBufRemoveCallback: platform.PublishFunc,
+			MaxHeapAllocBytes:   global.AgentConf.Buffer.MaxHeapAlloc,
+			MinBufSize:          global.AgentConf.Buffer.MinBufferSize,
+		}
 
-	// attach the buffer controller to the publisher
-	pub := publisher.NewPublisher(publisher.Config{}, bufCtrl)
-	pub.Start(wg)
+		buffer := buf.NewPriorityBuffer(global.AgentConf.Buffer.TTL)
+		bufCtrl := buf.NewController(bufCtrlConf, buffer)
 
-	ctx, cancel = context.WithCancel(context.Background())
-	// register Metrika Platform exporter
-	// TODO: make possible to disable in configuration
-	global.DefaultExporterRegisterer.Register(pub, ch)
+		// attach the buffer controller to the publisher
+		pub := publisher.NewPublisher(publisher.Config{}, bufCtrl)
+		pubCtx, pubCancel = context.WithCancel(context.Background())
+		pub.Start(pubCtx, wg)
+		subCh := newSubscriptionChan()
+		subscriptions = append(subscriptions, subCh)
+		global.DefaultExporterRegisterer.Register(pub, subCh)
+	}
 
-	// register other exporters (if enabled)
-	if global.AgentConf.Runtime.UseExporters {
-		exporters, err := contrib.GetExporters()
-		if err != nil {
-			log.Fatalw("could not setup the exporters", zap.Error(err))
-		}
-		for _, exporter := range exporters {
-			subCh := newSubscriptionChan()
-			subscriptions = append(subscriptions, subCh)
-			global.DefaultExporterRegisterer.Register(exporter, subCh)
+	for expName, exporterCfg := range global.AgentConf.Runtime.ExportersRaw {
```

Comment:
Similar to my comment about moving code from main: I think that's too much logic to put in main (there have no tests).

Reply:
Addressed in 1238bc5